src/EditorFeatures/CSharp/SignatureHelp/InvocationExpressionSignatureHelpProvider_MethodGroup.cs

```diff
@@ -43,7 +43,8 @@ internal partial class InvocationExpressionSignatureHelpProvider
                 }
 
                 var includeInstance = !throughExpression.IsKind(SyntaxKind.IdentifierName) ||
-                    semanticModel.LookupSymbols(throughExpression.SpanStart, name: throughSymbol.Name).Any(s => !(s is INamedTypeSymbol));
+                    semanticModel.LookupSymbols(throughExpression.SpanStart, name: throughSymbol.Name).Any(s => !(s is INamedTypeSymbol)) ||
+                    (!(throughSymbol is INamespaceOrTypeSymbol) && semanticModel.LookupSymbols(throughExpression.SpanStart, container: throughSymbol.ContainingType).Any(s => !(s is INamedTypeSymbol)));
```

Comment:
I've tried the same scenario with `List<SomeUndefinedType>` and it behaves as expected, and the closest I could get to your scenario was with the private field being `UndefinedType` but we don't get this far in that case because the field type doesn't bind, so I think we're correct in the assumption that we'll always get `ToString/GetHashCode/GetType` or not this far at all.